### ISSUE TITLE
nixos/pam: assemble via the module system

### ISSUE THREAD

**Issue by flokli** [NixOS/nixpkgs#90640]
`nixos/modules/security/pam.nix` contains a huge monolith of pam configuration.

This can be described in a more composable way by using the module system to assemble this configuration file.

While doing this, we'd need to determine a concept on how things should be ordered, taking into account different stages of the authentication process.

Helpful pointers:
 - http://www.rjsystems.nl/en/2100-pam-debian.php
 - https://manpages.debian.org/stretch/libpam-runtime/pam-auth-update.8.en.html

**Comment by rissson** [comment]
Is anyone working on this? Or should I get ready to get my hands dirty :D

**Comment by flokli** [comment]
Please do! :-)

**Comment by Rudi9719** [comment]
Hey @rissson how are you doing on this? Need a hand?

**Comment by rissson** [comment]
> Hey @rissson how are you doing on this? Need a hand?

I'm afraid I haven't started anything on this  :/ My current workaround is overriding the whole module with my requirements. However, if you want to work on this too, maybe we could find a time to meet and throw some ideas and maybe have a basic PoC, which we can then both work on!

**Comment by Rudi9719** [comment]
I'm not even sure where to begin; but as a former Gentoo user I'm comfortable with modifying PAM files etc. I'm looking to learn the Nix configurations. Are you on Keybase? I can reach out there and we can get this sorted. I'm going to start reading the links from the initial post

**Comment by rissson** [comment]
Yes, I'm on keybase (`risson`), but I don't check it often. Feel free to send me an email though, you should be able to find my email in my commits.

**Comment by kwohlfahrt** [comment]
@rissson I'd be interested in helping here - let me know if you run out of time to work on this or would like help with any part of it.

**Comment by rissson** [comment]
@kwohlfahrt sure! If I start anything anytime soon, I'll put it in a draft PR and link it here.

**Comment by aanderse** [comment]
@flokli i would suggest mission accomplished and this issue can be closed now

**Comment by flokli** [comment]
I somehow missed the PR, but yes, indeed this seems fixed.. Thanks everyone!
